packages/grid/x-data-grid/src/constants/gridClasses.ts

```diff
@@ -382,7 +394,7 @@ export function getDataGridUtilityClass(slot: string): string {
   return generateUtilityClass('MuiDataGrid', slot);
 }
 
-export const gridClasses = generateUtilityClasses('MuiDataGrid', [
+export const gridClasses = generateUtilityClasses<GridClassKey>('MuiDataGrid', [
```

Comment:
Adding the generic to make sure that TS do not accept adding a class here but not on the interface